### PR TITLE
WIP: fixed price sale with admin

### PR DESCRIPTION
**Pull request by emmanueldenloye**
Please note that the admin account in the marketplace test is an
account that is not included in the sandbox. This has to be manually
created. If you wish to run this test, please consult me (Emmanuel) for
extra details.

### CODE REVIEW

File: contracts/ligo/src/fixed_price_sale_market_tez.mligo [hunk 1]
```diff
@@ -58,29 +65,40 @@ let transfer_tez (price, seller : tez * address) : operation =
...
-  | Buy sale -> buy_token(sale, storage)
-  | Cancel sale -> cancel_sale(sale,storage)
+  | Sell sale ->
+     let u = fail_if_paused(storage.admin) in
+     deposit_for_sale(sale.sale_token_param_tez, sale.sale_price, storage)
+  | Buy sale ->
+     let u = fail_if_paused(storage.admin) in
+     buy_token(sale, storage)
+  | Cancel sale ->
+     let u = fail_if_not_admin storage.admin in
```
Comment (mycodecrafting) [inline contracts/ligo/src/fixed_price_sale_market_tez.mligo:98]: why can only an admin cancel marketplace sales?

File: contracts/ligo/src/fixed_price_sale_market_tez.mligo [hunk 2]
```diff
@@ -58,29 +65,40 @@ let transfer_tez (price, seller : tez * address) : operation =
...
       transfer_nft (sale_token.token_for_sale_address, sale_token.token_for_sale_token_id, Tezos.sender, Tezos.self_address) in
     let sale_param = { sale_seller = Tezos.sender; sale_token = sale_token } in
-    let new_s = Big_map.add sale_param price storage in
+    let new_s = { storage with sales = Big_map.add sale_param price storage.sales } in
     (transfer_op :: []), new_s
 
-let cancel_sale(sale, storage: sale_param_tez * storage) : (operation list * storage) = match Big_map.find_opt sale storage with
+let cancel_sale(sale, storage: sale_param_tez * storage) : (operation list * storage) = match Big_map.find_opt sale storage.sales with
     | None -> (failwith "NO_SALE" : (operation list * storage))
     | Some price -> if sale.sale_seller = Tezos.sender then
```
Comment (mycodecrafting) [inline contracts/ligo/src/fixed_price_sale_market_tez.mligo:85]: To cancel you must both be an admin and the seller? Shouldn't the logic be, an admin OR the seller can cancel?

Comment (emmanueldenloye) [reply contracts/ligo/src/fixed_price_sale_market_tez.mligo:85]: Ah, good catch. Thanks. I'll change the logic accordingly.

Comment (EGuenz) [reply contracts/ligo/src/fixed_price_sale_market_tez.mligo:85]: Shouldn't only admin be able to make a sale as well? Then the logic only needs to be, only a seller (who also is necessarily the admin) can cancel.

**Comment by emmanueldenloye** [comment]
I'm going to be closing this PR. All of these changes (along with some others) will be featured in another branch integrating all of the contracts that both Eli and I have written.

**Pull request closed without merging**